File: 8-kyu/tip-calculator/python/solution.py
from math import ceil

def calculate_tip(amount, rating):
    rating_lookup = {'terrible': 0, 'poor': .05, 'good': .1, 'great': .15, 'excellent': .2}
    rating = rating.lower()
    return 'Rating not recognised' if rating not in rating_lookup else ceil(amount * rating_lookup[rating])

File: 8-kyu/tip-calculator/javascript/solution.js
const calculateTip = (amount, rating) => {
  const percentages = {
    terrible: 0,
    poor: 0.05,
    good: 0.1,
    great: 0.15,
    excellent: 0.2
  };
  const ratingLower = rating.toLowerCase();
  return percentages.hasOwnProperty(ratingLower) ? Math.ceil(amount * percentages[ratingLower]) : "Rating not recognised";
};
